src/main/kotlin/parts/wisdom/arcraider/visualization/ARCWindow.kt

```diff
@@ -3,48 +3,57 @@ package parts.wisdom.arcraider.visualization
 import parts.wisdom.arcraider.SerializedGrid
 import parts.wisdom.arcraider.Task
 import parts.wisdom.arcraider.loadTaskFromFile
+import java.awt.BorderLayout
 import java.awt.Dimension
 import javax.swing.Box
 import javax.swing.BoxLayout
-import javax.swing.JButton
-import javax.swing.JFileChooser
 import javax.swing.JFrame
 import javax.swing.JLabel
 import javax.swing.JPanel
 import javax.swing.JToggleButton
+import kotlin.math.max
 
 private const val GAP_SIZE = 12
-private const val GRID_SIZE = 40
+private const val GRID_SIZE = 400
 
-typealias TaskGrids = Pair<List<GridPane>, List<GridPane>>
+typealias TaskGrids = List<Pair<GridPane, GridPane>>
 
-class ARCWindow(width: Int, height: Int, title: String, task: Task) : JFrame() {
+class ARCWindow(title: String, task: Task) : JFrame() {
 
-    private var taskType = TaskType.TEST
+    private var taskType = TaskType.TRAIN
     private val content = JPanel()
 
     init {
-        createUI(width, height, title, task)
+        createUI(title, task)
     }
 
-    private fun createUI(width: Int, height: Int, title: String, task: Task) {
+    private fun createUI(title: String, task: Task) {
         setTitle(title)
         defaultCloseOperation = EXIT_ON_CLOSE
-        setSize(width, height)
         setLocationRelativeTo(null)
         content.layout = BoxLayout(content, BoxLayout.Y_AXIS)
-        add(content)
+
+        val root = JPanel().apply {
+            layout = BorderLayout()
+            add(
+                    FileBrowser(System.getProperty("user.home")) {
+                        refresh(loadTaskFromFile(it))
+                    },
+                    BorderLayout.WEST
+            )
+            add(content, BorderLayout.CENTER)
+        }
+        add(root)
 
         renderTask(task)
+        pack()
     }
 
     private fun renderTask(task: Task) {
         content.apply {
             add(Box.createRigidArea(Dimension(0, GAP_SIZE)))
             add(
                     Box.createHorizontalBox().apply {
-                        add(Box.createRigidArea(Dimension(GAP_SIZE, 0)))
-                        add(createBrowseButton())
                         add(Box.createRigidArea(Dimension(GAP_SIZE, 0)))
                         add(createToggle(task, TaskType.TEST))
```

Comment:
To me, it seems strange to have the TEST button first, because I think of the training examples coming before the test input.

Reply:
I swapped them